### PR TITLE
Add offset iron sights

### PR DESCRIPTION
**Pull request by mugling**
Now that we have `DISABLE_SIGHTS` aiming at close range via `rifle_scope` can be slow as the base weapon sights are obscured. IRL this problem is solved via offset iron sights.

a1

`offset_sights` are slightly inferior to regular sights and use the `rail` slot where they compete with the quicker but less accurate `rail_laser_sight`.

### CODE REVIEW

File: data/json/items/gunmod/sights.json [hunk 1]
```diff
@@ -37,14 +37,14 @@
   {
     "id": "improve_sights",
     "type": "GUNMOD",
-    "name": "improved iron sights",
+    "name": "iron sights",
```
Comment (BorkBorkGoesTheCode) [inline data/json/items/gunmod/sights.json]: Does this mean guns start without iron sights?

Comment (Rivet-the-Zombie) [inline data/json/items/gunmod/sights.json]: Yeah, the 'improved' part of the name implies that they're a step up from what's already there. This makes it seem more like we're adding sights to a gun that has none.

File: data/json/items/gunmod/rail.json [hunk 2]
```diff
@@ -25,6 +25,19 @@
     "flags": [ "STR_RELOAD" ]
   },
   {
+    "id": "offset_sights",
+    "copy-from": "improve_sights",
+    "type": "GUNMOD",
+    "name": "offset iron sights",
+    "name_plural": "offset iron sights",
+    "description": "An alternative set of iron sights mounted at 45° for use when a scope or other modification prevents use of the primary sights",
```
Comment (Night-Pryanik) [inline data/json/items/gunmod/rail.json]: Please add period at the end.

File: data/json/items/gunmod/sights.json (hunk 1 above)
Comment (mugling) [inline data/json/items/gunmod/sights.json]: Change here is for consistency with `name_plural`. At present this item offers no improvement over standard sights but they are currently useful when a less conventional (especially home made) has poor sights. 

Whatever we go with `name` and `name_plural` should be equivalent. If we want them to be an improvement then we can look at that in a separate PR.

File: data/json/items/gunmod/rail.json (hunk 2 above)
Comment (mugling) [inline data/json/items/gunmod/rail.json]: Done

**Comment by Coolthulhu** [comment]
Why even have alternative sights, though?
What is exactly the point of obscuring sights?

**Comment by mugling** [comment]
> What is exactly the point of obscuring sights?

Adding a `rifle_scope` blocks use of the base sights (see picture). This makes installing a `rifle_scope` more of a tradeoff. Offset sights provide a partial mitigation but prevent use of the equally useful `rail_laser_sight`

> Why even have alternative sights, though?

As in `improved_sights`. For guns that have even worse sights? We could deprecate them

**Pull request merged**